diff --git a/node_modules/vue-template-compiler/build.js b/node_modules/vue-template-compiler/build.js
index 0cc0eab..1f718d9 100644
--- a/node_modules/vue-template-compiler/build.js
+++ b/node_modules/vue-template-compiler/build.js
@@ -416,23 +416,22 @@ function parseHTML(html, options) {
         else {
             let endTagLength = 0;
             const stackedTag = lastTag.toLowerCase();
-            const reStackedTag = reCache[stackedTag] ||
-                (reCache[stackedTag] = new RegExp('([\\s\\S]*?)(</' + stackedTag + '[^>]*>)', 'i'));
-            const rest = html.replace(reStackedTag, function (all, text, endTag) {
-                endTagLength = endTag.length;
-                if (!isPlainTextElement(stackedTag) && stackedTag !== 'noscript') {
-                    text = text
-                        .replace(/<!\--([\s\S]*?)-->/g, '$1') // #7298
-                        .replace(/<!\[CDATA\[([\s\S]*?)]]>/g, '$1');
-                }
-                if (shouldIgnoreFirstNewline(stackedTag, text)) {
-                    text = text.slice(1);
-                }
-                if (options.chars) {
-                    options.chars(text);
-                }
-                return '';
-            });
+            const endTagStart = html.toLowerCase().indexOf('</' + stackedTag);
+      const endTagEnd = endTagStart < 0 ? -1 : html.indexOf('>', endTagStart + stackedTag.length + 2);
+      let text = endTagStart < 0 ? html : html.slice(0, endTagStart);
+      const rest = endTagEnd < 0 ? '' : html.slice(endTagEnd + 1);
+      endTagLength = endTagEnd < 0 ? 0 : endTagEnd + 1 - endTagStart;
+      if (!isPlainTextElement(stackedTag) && stackedTag !== 'noscript') {
+          text = text
+              .replace(/<!\--([\s\S]*?)-->/g, '$1') // #7298
+              .replace(/<!\[CDATA\[([\s\S]*?)]]>/g, '$1');
+      }
+      if (shouldIgnoreFirstNewline(stackedTag, text)) {
+          text = text.slice(1);
+      }
+      if (options.chars) {
+          options.chars(text);
+      }
             index += html.length - rest.length;
             html = rest;
             parseEndTag(stackedTag, index - endTagLength, index);
